.github/workflows/server-benchmark.yml

```diff
@@ -24,17 +26,93 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-gradle-
-    # JDK is needed to generate code
+      # Pinned to the commit hash of v1.3.0
+    - uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+      with:
+        sharedKey: ${{ runner.os }}-${{ env.rust_version }}-${{ github.job }}
+        target-dir: ./target
     - name: Set up JDK
       uses: actions/setup-java@v1
       with:
         java-version: ${{ env.java_version }}
-    # Install Rust
-    - uses: actions-rs/toolchain@v1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
       with:
         toolchain: ${{ env.rust_version }}
+        components: ${{ env.rust_toolchain_components }}
         default: true
     - name: Run integration tests
       run: |
         cd rust-runtime/aws-smithy-http-server/examples && \
           make && cargo test
+
+  run-benchmark:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      name: Gradle Cache
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+    - uses: actions/cache@v2
+      name: Benchmarks Cache
+      with:
+        path: ~/.wrk-api-bench
+        key: ${{ runner.os }}-wrk-api-bench-${{ hashFiles('**/*.json') }}
+        restore-keys: |
+          ${{ runner.os }}-wrk-api-bench-
+      # Pinned to the commit hash of v1.3.0
+    - name: Rust Cache
+      uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+      with:
+        sharedKey: ${{ runner.os }}-${{ env.rust_version }}-${{ github.job }}
+        target-dir: ./target
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ env.java_version }}
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.rust_version }}
+        components: ${{ env.rust_toolchain_components }}
+        default: true
+    - name: Install benchmarks dependencies
+      run: sudo apt-get install -y ${{ env.apt_dependencies }}
+    # Ubuntu 20.04 doesn't have wrk packaged, hence we need to build it 🤦
+    # This will go away as soon as github supports Ubuntu 21.04.
+    - name: Install wrk
+      run: |
+        git clone https://github.com/wg/wrk.git /tmp/wrk-build && \
+          cd /tmp/wrk-build && make -j8 wrk && sudo cp wrk /usr/local/bin
+    - name: Run benchmark
+      id: run-benchmark
+      run: |
+        # run the benchmark.
+        cd rust-runtime/aws-smithy-http-server/examples && \
+          make && RUN_BENCHMARKS=1 cargo test --release
+
+        # uncomment this for debugging purpose. it will print out the
+        # content of all the benchmarks found in the cache + the last one
+        # produced by the current run.
+        # for x in ~/.wrk-api-bench/*; do echo "Benchmark $x content:"; cat $x | jq .; echo; done
```

Comment:
```suggestion
        # for x in ~/.wrk-api-bench/*; do echo "Benchmark $x content:"; jq . "$x"; echo; done
```

Reply:
I am never ashamed of my unnecessary cat!